enclave/impl/src/message_bus/marshalled_record.rs

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2018-2023 The MobileCoin Foundation
+
+//! A marshalled record
+
+use super::{MessageBusValueSize, MsgId};
+use aligned_cmov::{
+    typenum::{U32, U64, U72, U8},
+    A8Bytes, AsNeSlice,
+};
+use generic_array::sequence::Split;
+use mc_bomb_enclave_api::QueryError;
+use mc_bomb_types::Record;
+
+/// A record marshalled to the layout of a value in the oblivious message bus.
+///
+/// The layout is:
+/// 0..32 = sender bytes
+/// 32..64 = recipient bytes
+/// 64..72 = timestamp bytes (native endian)
+/// 72...  = payload bytes
+///
+/// The total size is determined by the MessageBusValueSize constant, which is
+/// determined by the choice of ORAM parameters, and the size of the message bus
+/// key.
+///
+/// The payload, and record, are always fixed size given these parameters.
+pub struct MarshalledRecord {
+    data: A8Bytes<MessageBusValueSize>,
+}
```

Comment:
I think this should be trait rather than a type, this will allow to avoid some evil casts and unnecessary copies when reading out of the ring buffer